packages/core/contracts/cross-chain-oracle/OracleSpoke.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../oracle/interfaces/OracleAncillaryInterface.sol";
+import "../oracle/interfaces/RegistryInterface.sol";
+import "./OracleBase.sol";
+import "../common/implementation/AncillaryData.sol";
+import "../common/implementation/Lockable.sol";
+import "./ChildMessengerInterface.sol";
+
+/**
+ * @title This contract is primarily intended to receive messages on the child chain from a parent chain and allow
+ * contracts deployed on the child chain to interact with this contract as an Oracle. Moreover, this contract gives
+ * child chain contracts the ability to trigger cross-chain price requests to the mainnet DVM. This Spoke knows how
+ * to communicate with the parent chain via a "ChildMessenger" contract which directly communicates with the
+ * "ParentMessenger" on mainnet.
+ * @dev The intended client of this contract is an OptimisticOracle on sidechain that needs price
+ * resolution secured by the DVM on mainnet.
+ */
+contract OracleSpoke is OracleBase, OracleAncillaryInterface, Lockable {
+    ChildMessengerInterface public messenger;
+
+    event SetChildMessenger(address indexed childMessenger);
+
+    constructor(address _finderAddress, ChildMessengerInterface _messengerAddress) OracleBase(_finderAddress) {
+        messenger = _messengerAddress;
+        emit SetChildMessenger(address(messenger));
+    }
+
+    // This assumes that the local network has a Registry that resembles the mainnet registry.
+    modifier onlyRegisteredContract() {
+        RegistryInterface registry = RegistryInterface(finder.getImplementationAddress(OracleInterfaces.Registry));
+        require(registry.isContractRegistered(msg.sender), "Caller must be registered");
+        _;
+    }
+
+    modifier onlyMessenger() {
+        require(msg.sender == address(messenger), "Caller must be messenger");
+        _;
+    }
+
+    /**
+     * @notice This is called to bridge a price request to mainnet. This method will enqueue a new price request
+     * or return silently if already requested. Price requests are relayed to mainnet (the "Parent" chain) via
+     * the ChildMessenger contract.
+     * @dev Can be called only by a registered contract that is allowed to make DVM price requests. Will mark this
+     * price request as Requested, and therefore able to receive the price resolution data from mainnet.
+     * @param identifier Identifier of price request.
+     * @param time Timestamp of price request.
+     * @param ancillaryData extra data of price request.
+     */
+    function requestPrice(
```

Comment:
Do you see any danger in _not_ making these backwards compatible with the DVM methods that _don't_ have ancillary data?

Put differently, are there any contracts someone might want to deploy that this would break? EMP, Perp?

cc @chrismaree

Reply:
I think at this point we should try move all DVM contracts to using ancillary data interfaces. ideally in the near future (next audit cycle perhaps?) we can deprecate the old interfaces in favour of the ancillary data-only ones.

However, you do raise a good point on how this _could_ break some things. The latest EMP and Perp both have ancillary data support so it won't be an issue with those. it _could_ be an issue with legacy contracts but we dont want to support those anyways.

Reply:
How would we make it backwards compatible--overload the `requestPrice` method?

Reply:
> The latest EMP and Perp both have ancillary data support

I don't think this is true.

See EMP: https://github.com/UMAprotocol/protocol/blob/2fc8fe4237156e68a02bf15ff9f981eed0598137/packages/core/contracts/financial-templates/expiring-multiparty/PricelessPositionManager.sol#L837

See Perp: https://github.com/UMAprotocol/protocol/blob/2fc8fe4237156e68a02bf15ff9f981eed0598137/packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol#L658


So the question is whether we want to support these contracts short term. We could also update them and get that into this audit, but that will cause some real test churn I suspect.

Reply:
Well since we're not concerned about bytecode bloat let's overload the `requestPrice` method?

Reply:
@mrice32 that's a great point and is really irritating...both of those contracts _use_ ancillary data but also use the overloaded methods.

@nicholaspai I think the simplest thing to do is to overload `requestPrice` that just defaults ancillary data to blank, the same way the `Voting.sol` contracts do this to provide backwards compatibility.

Reply:
Done I've made `OracleSpoke` extend the `OracleInterface` (in addition to `OracleAncillaryInterface`) and overloaded `requestPrice`, `hasPrice`, `getPrice`